File: 2965-find-missing-and-repeated-values/2965-find-missing-and-repeated-values.py
class Solution:
    def findMissingAndRepeatedValues(self, grid: List[List[int]]) -> List[int]:
        cnt = {}
        n = len(grid)
        for row in grid:
            for i in row:
                cnt[i] = cnt.get(i, 0) + 1
        a, b = 0, 0
        for i in range(1, n*n+1):
            if i not in cnt: b = i
            elif cnt[i] == 2: a = i
        return [a, b]

File: 2965-find-missing-and-repeated-values/2965-find-missing-and-repeated-values.cpp
class Solution {
public:
    vector<int> findMissingAndRepeatedValues(vector<vector<int>>& grid) {
        int n = grid.size();
        vector<int> cnt(n*n, 0);
        vector<int> ans(2);
        for(const auto &row: grid) {
            for(const auto &i: row) {
                cnt[i-1]++;
            }
        }
        for(int i = n*n; i >= 1; i--) {
            if(cnt[i-1] == 0) ans[1] = i;
            if(cnt[i-1] == 2) ans[0] = i;
        }
        return ans;
    }
};